onnxruntime/core/providers/dml/DmlExecutionProvider/src/ErrorHandling.h

```diff
@@ -11,3 +31,52 @@
              THROW_HR(StatusCodeToHRESULT(static_cast<onnxruntime::common::StatusCode>(_status.Code())));       \
         }                                                                                                       \
     } while (0)
+#endif
+
+#ifdef ORT_NO_EXCEPTIONS
+#define ORT_THROW_IF_FAILED(hr) \
+    if(!SUCCEEDED(hr))          \
+    {                           \
+        ORT_THROW(hr);          \
+    }
+#else
+#define ORT_THROW_IF_FAILED(hr) THROW_IF_FAILED(hr)
+#endif
+
+#ifdef ORT_NO_EXCEPTIONS
+#define ORT_THROW_LAST_ERROR_IF_NULL(ptr) \
+    if(ptr == nullptr)          \
+    {                           \
+        ORT_THROW(E_INVALIDARG);          \
```

Comment:
>E_INVALIDARG [](http://example.com/codeflow?start=18&length=12)

E_POINTER is more representative (not that it much matters since the HRESULT won't actually be thrown in the NO_EXCEPTIONS case anyway).